### PR TITLE
Update /tutorial to include `!` suffix

### PR DESCRIPTION
**Pull request by rtfeldman**
Also made a variety of other changes!

### CODE REVIEW

File: www/content/tutorial.md [hunk 1]
```diff
@@ -1586,154 +1570,276 @@ app "cli-tutorial"
...
+    input =
+        Stdin.line!
+        |> Task.mapErr ReadStdin
+```
+
+This version starts with the `Stdin.line` task, awaits it with `!` to get its `Str` value if it succeded, and then tries to call `Task.mapErr` on that `Str` value. This will be a type mismatch!
+
+In general, an easy rule to remember is "do the `!` last" - if you want to modify a `Task` before running it, use the `!` only after you've made all the modifications and ended up with the final `Task` you'd like to run.
 
 See the [Task & Error Handling example](https://www.roc-lang.org/examples/Tasks/README.html) for a more detailed explanation of how to use tasks to help with error handling in a larger program.
```
Comment (lukewilliamboswell) [inline www/content/tutorial.md:1831]: Needs updating to use `!` suffix --> ready at https://github.com/roc-lang/examples/pull/172

File: www/content/tutorial.md [hunk 2]
```diff
@@ -1586,154 +1570,276 @@ app "cli-tutorial"
...
 
+### [Displaying Roc values with `Inspect.toStr`](#inspect) {#inspect}
+
+The [`Inspect.toStr`](https://www.roc-lang.org/builtins/Inspect#toStr) function returns a `Str` representation of any Roc value. It's useful for things like debugging and logging (although [`dbg`](https://www.roc-lang.org/tutorial#debugging) is often nicer for debugging in particular), but its output is almost never something that should be shown to end users! In this case we're just using it for our own learning, but it would be better to run a `when` on `e` and display a more helpful message.
```
Comment (lukewilliamboswell) [inline www/content/tutorial.md:1835]: Might be worthwhile linking to https://www.roc-lang.org/abilities#inspect-ability

Also jus noticed that has been merged, and now also needs to be updated. Issue added https://github.com/roc-lang/roc/issues/6679

**Review by lukewilliamboswell — COMMENTED** [review]
Looking good 😃

File: www/content/tutorial.md [hunk 3]
```diff
@@ -1586,154 +1570,276 @@ app "cli-tutorial"
...
 
-The type of `Task.await` is:
+Although this default error handling behavior might be exactly what we want if we're writing a quick script,
+high-quality programs handle errors gracefully. Fortunately, we can do this nicely in Roc!
+
+If we wanted to add the type annotation to `main` that Roc is inferring for it, we would add this annotation:
 
 ```roc
-Task.await : Task a err, (a -> Task b err) -> Task b err
+main : Task {} [Exit I32, StdoutErr Stdout.Err, StinErr Stdin.Err]
```
[the remark this answers is not in the document]
Comment (lukewilliamboswell) [reply www/content/tutorial.md:1598]: basic-cli has been updated in https://github.com/roc-lang/basic-cli/pull/188, no need to change this PR

File: www/content/tutorial.md [hunk 4]
```diff
@@ -1532,9 +1516,9 @@ See the [Ingest Files Example](https://www.roc-lang.org/examples/IngestFiles/REA
 
 ## [Tasks](#tasks) {#tasks}
 
-Tasks are technically not part of the Roc language, but they're very common in platforms. Let's continue using the [basic-cli](https://github.com/roc-lang/basic-cli) platform we've been using up to this point as an example!
+Tasks are not currently part of the Roc builtins—currently, each platform exposes their own `Task` implementation, but the plan is to standardize them into a builtin `Task` like module like the builtin modules we already have for `List`, `Str`, and so on—but they're an important part of building Roc applications, so let's continue using the [basic-cli](https://github.com/roc-lang/basic-cli) platform we've been using up to this point as an example!
```
Comment (agu-z) [inline www/content/tutorial.md:1519]: Did you mean to have "currently" twice here?

Comment (rtfeldman) [reply www/content/tutorial.md:1519]: Nope, fixed!

File: www/content/tutorial.md [hunk 5]
```diff
@@ -1586,154 +1570,276 @@ app "cli-tutorial"
...
-    _ <- await (Stdout.line "Type something press Enter:")
-    input <- await Stdin.line
+    Stdout.line "Type something and press Enter."
+    |> Task.mapErr! PrintPrompt
+
+    input = Stdin.line!
 
-    Stdout.line "Your input was: $(Inspect.toStr input)"
+    Stdout.line "You entered: $(input)"
+    |> Task.mapErr! PrintInput
```
[the remark this answers is not in the document]
Comment (rtfeldman) [reply www/content/tutorial.md:1788]: Yeah it's a good question! Seems like something where the style will evolve and conventions will emerge over time. 😄

File: www/content/tutorial.md [hunk 6]
```diff
@@ -1586,154 +1570,276 @@ app "cli-tutorial"
...
+
+Basically, this function creates a task which runs one task, and then runs a second task which can use the output of the first task. (If the first task fails, the second task never gets run.)
 
-Stdout.line "Your input was: $(Inspect.toStr input)"
+More specifically, `Task.await` returns a `Task` which:
+1. Runs the given `Task a err`
+2. If it fails with some error, returns a `Task` which fails with that same error. (So if the given `Task a err` fails, the `a -> Task b err` function never gets called.)
+3. If it succeeds (meaning it produces a success value which has the type `a`), pass the value it succeeded with to the `a -> Task b err` function. Whatever `Task b err` that function returns will be the `Task b err` the entire `Task.await` call returns.
+
+The `!` suffix is syntax sugar for connecting tasks using `Task.await`. Let's revisit our earlier example here:
```
[the remark this answers is not in the document]
Comment (rtfeldman) [reply www/content/tutorial.md:1743]: Feels slightly distracting here, but I appreciate the idea!

**Pull request merged**